python/tskit/trees.py

```diff
@@ -1605,8 +1711,20 @@ def nodes(self, root=None, order="preorder"):
         roots = [root]
         if root is None:
             roots = self.roots
-        for u in roots:
-            yield from iterator(u)
+        if order == "minlex_postorder" and len(roots) > 1:
+            # we need to visit the roots in minlex order as well
+            # we first visit all the roots and then sort by the min value
```

Comment:
This multi-root case is not documented prominently in the nodes() docstring, should we highlight it? It's unlikely that users would be using the function in the multiple roots case.

Reply:
I don't think we need to worry about it, as you say.